sympy/functions/elementary/hyperbolic.py

```diff
@@ -1873,6 +1939,44 @@ def eval(cls, arg):
         if arg.could_extract_minus_sign():
             return -cls(-arg)
 
+    @staticmethod
+    @cacheit
+    def taylor_term(n, x, *previous_terms):
+        if n == 0:
+            return log(2 / x)
+        elif n < 0 or n % 2 == 1:
+            return S.Zero
+        else:
+            x = sympify(x)
+            if len(previous_terms) > 2 and n > 2:
+                p = previous_terms[-2]
+                return -p * ((n - 1)*(n-2)) * x**2/(4 * (n//2)**2)
+            else:
+                k = n // 2
+                R = RisingFactorial(S.Half, k) *  n
+                F = factorial(k) * n // 2 * n // 2
+                return S.NegativeOne**(k +1) * R / F * x**n / 4
+
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):  # acsch
+        arg = self.args[0]
+        x0 = arg.subs(x, 0).cancel()
+        # Handling Branch cuts (-I, I)
+        if x0 in (-S.ImaginaryUnit, S.ImaginaryUnit, S.Zero):
+            return self.rewrite(log)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+        if x0 is S.ComplexInfinity:
+            return (1/arg).as_leading_term(x)
+        ndir = arg.dir(x, cdir if cdir else 1)
+        if (1 + x0**2).is_positive:
```

Comment:
`x0.is_imaginary and (1 + x0**2).is_positive` to exclude real values of `x0`.

Reply:
Yeah this should be added, if we look into the block , real values of `x0` won't make a difference
```
        if (1 + x0**2).is_positive:
            if re(ndir).is_positive:
                if im(x0).is_positive:
                    return -self.func(x0) - I*pi
```
But as our primary focus is branch cuts taking execution till the last line shouldn't be the case and rather we should reject such case at the first if condition only !